### PR TITLE
Validations and model specs.

### PR DESCRIPTION
**Pull request by Fene-87**
### In this milestone, I add validations to your models and create specs for them. I have added the following validations:
- Name must not be blank in the User model.
- PostsCounter must be an integer greater than or equal to zero in the User model.
- Title must not be blank and must not exceed 250 characters in the Post model.
- CommentsCounter must be an integer greater than or equal to zero in the Post model.
- LikesCounter must be an integer greater than or equal to zero in the Post model.

### CODE REVIEW

File: spec/models/user_spec.rb
Comment (fobadara) [inline spec/models/user_spec.rb:1]: [REQUIRED] It is a requirement to include tests for custom methods. Kindly include tests for recent_posts

File: spec/models/post_spec.rb
Comment (fobadara) [inline spec/models/post_spec.rb:1]: [REQUIRED] Kindly include tests for recent_comments and update_post_counter as this is a requirement.

File: spec/models/comment_spec.rb
Comment (fobadara) [inline spec/models/comment_spec.rb:1]: [REQUIREMENT] It is a requirement to test custom methods. Kindly test update_comments_counter

File: spec/models/like_spec.rb
Comment (fobadara) [inline spec/models/like_spec.rb:1]: [REQUIREMENT] It is a requirement to test custom methods. Kindly test update_likes_counter

**Review by fobadara — CHANGES_REQUESTED** [review]
Hi @Fene-87, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

## Highlights
✅ I like the fact that you are testing all validations including those that are not from the requirements

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/comment_spec.rb
Comment (Fene-87) [reply spec/models/comment_spec.rb:1]: done

File: spec/models/user_spec.rb
Comment (Fene-87) [reply spec/models/user_spec.rb:1]: done

File: spec/models/like_spec.rb
Comment (Fene-87) [reply spec/models/like_spec.rb:1]: done

File: spec/models/post_spec.rb
Comment (Fene-87) [reply spec/models/post_spec.rb:1]: done

**Comment by Fene-87** [comment]
Hey @fobadara I have now implemented all the tests for custom methods.

**Review by TadesseDev — APPROVED** [review]
### Hi @Fene-87 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
giphy (1)



## Highlights
 - [x] Good job making the changes requested by the previous code reviewer. ✅
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**